apps/system/test/apps/wifi-direct/js/wifip2p.js

```diff
@@ -0,0 +1,147 @@
+/* -*- Mode: js; js-indent-level: 2; indent-tabs-mode: nil -*- */
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* global dump */
+
+'use strict';
+
+(function(exports) {
+  var gWpsMethod = 'pbc';
+  var gGoIntent = '1';
+
+  function debug(s) {
+    dump('Wifi Direct Test: ' + s);
+  }
+
+  function WifiP2pManager(aPeerListView) {
+    this.manager = navigator.mozWifiP2pManager;
+    this.peerListView = aPeerListView;
+  }
+
+  WifiP2pManager.prototype = {
+    init: function() {
+      if (!this.manager) {
+        debug('Wifi Direct not supported.');
+        return false;
+      }
+
+      this.manager.setScanEnabled(true);
+
+      // Add peerListView controller.
+      this.peerListView.onPeerClicked = aPeerInfo => {
```

Comment:
We can simply do: `this.peerListView.onPeerClicked = this.onPeerClicked.bind(this);`